src/compiler/factory.ts

```diff
@@ -21,11 +21,13 @@ namespace ts {
     }
 
     export function updateNode<T extends Node>(updated: T, original: T): T {
-        setOriginalNode(updated, original);
-        if (original.startsOnNewLine) {
-            updated.startsOnNewLine = true;
+        if (updated !== original) {
+            setOriginalNode(updated, original);
+            if (original.startsOnNewLine) {
+                updated.startsOnNewLine = true;
+            }
+            aggregateTransformFlags(updated);

```

Comment:
We always call `aggregateTransformFlags` at some point during the transformation pass, usually just after this function is called. By calling it here we cut down on how deep `aggregateTransformFlags` has to analyze a subtree.